packages/salesforcedx-utils-vscode/src/helpers/utils.ts

```diff
@@ -24,6 +24,7 @@ export function extractJsonObject(str: string): any {
 // To get around this, fs.realpathSync.native() is called to get the
 // URI with the actual file name.
 export function flushFilePath(filePath: string): string {
+  // filePath is in the format of "/Users/{user-name}/{path-to-apex-file.cls}"
```

Comment:
I'm not sure this comment is valuable.  This is only true on mac, and only if the sf project is under the users home directory (it could be anywhere on the filesystem) It could be misleading to our future selves reading it.

Reply:
I had the opposite reaction of "oh how useful!" 😄 But I get your point here.

Reply:
When I first discovered the bug and I was stepping through the code, it was not obvious what the issue was, which was, when `fs.realpathSync.native` is called, it's sometimes called with a path of `file://Users/...` and sometimes with a. path of `/Users/...`, and when passed a string with "file:" in it, it would thrown an exception.

Knowing what I do now, yea, it's obvious, but when I was initially looking into it, it wasn't obvious.

OK, so why use paths with "file:" in them at all?  We have code (like what's in testOutlineProvider.ts) which relies on it.  I'm not aware of the historical reasons why this is, but would prefer to not change additional components/areas with this PR.

Reply:
That's cool.  In an ideal world we should create types and remove the mental type checking of strings [ref](https://stackoverflow.com/questions/51445767/how-to-define-a-regex-matched-string-type-in-typescript).  In a less ideal world we should validate the filePath parameter matches what we expect and throw an error if not correct. 

 The part I take issue with here is `/Users/{user-name}/`.  It's inaccurate and could lead devs down a bad assumption path about what to expect from the input parameter.

Reply:
_"The part I take issue with here is /Users/{user-name}/..."_

I agree.  I still need to run this on Windows an verify the fix works.  (I only have a VM for this, and it's dog slow and I've been swamped this week and haven't gotten to this yet).  I'll post an update here with what I find.  I know the format will be different (eg `/Users/{user-name}/...` vs `C:/Users/{user-name}/...` so I will probably document both expected Mac and Windows paths in both locations.

Reply:
Randi ran the branch on her Windows machine this morning it ran w/o any issues (the file paths worked).  I added comments on what the expected path for Mac OSX and Windows are expected.